File: Android/app/src/main/java/com/sans347/app/ui/screens/GraphsScreen.kt
package com.sans347.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.sans347.app.data.GraphConfig
import com.sans347.app.data.allGraphs
import com.sans347.app.ui.graph.AspectFitGraphFrame
import com.sans347.app.ui.graph.LandscapeFigureNavRow
import com.sans347.app.ui.graph.LandscapeGraphSidebar
import com.sans347.app.ui.graph.Sans347Graph
import com.sans347.app.ui.graph.ZoomableGraphBox
import com.sans347.app.ui.theme.SansColors

@Composable
fun GraphsScreen(
    currentIndex: Int,
    onIndexChange: (Int) -> Unit,
    immersiveGraph: Boolean,
    modifier: Modifier = Modifier,
) {
    val graph = allGraphs[currentIndex]

    if (immersiveGraph) {
        GraphsImmersiveLayout(
            graph = graph,
            currentIndex = currentIndex,
            onIndexChange = onIndexChange,
            modifier = modifier,
        )
        return
    }

    BoxWithConstraints(
        modifier = modifier
            .fillMaxSize()
            .background(SansColors.White),
    ) {
        Column(
            modifier = Modifier
                .widthIn(max = 896.dp)
                .fillMaxWidth()
                .align(Alignment.TopCenter)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 16.dp)
                .padding(bottom = 100.dp),
        ) {
            GraphHeaderCard(graph, currentIndex, onIndexChange)

            Spacer(Modifier.height(16.dp))

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(SansColors.White, RoundedCornerShape(12.dp))
                    .border(1.dp, SansColors.Gray200, RoundedCornerShape(12.dp))
                    .padding(8.dp),
            ) {
                ZoomableGraphBox(modifier = Modifier.fillMaxWidth()) {
                    Sans347Graph(config = graph, modifier = Modifier.fillMaxWidth())
                }
            }

            Spacer(Modifier.height(16.dp))

            Text(
                graph.applicationText,
                fontSize = 14.sp,
                color = SansColors.Gray700,
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .fillMaxWidth()
                    .background(SansColors.White, RoundedCornerShape(12.dp))
                    .border(1.dp, SansColors.Gray200, RoundedCornerShape(12.dp))
                    .padding(16.dp),
            )

            Spacer(Modifier.height(12.dp))

            Text(
                graph.footerText,
                fontSize = 14.sp,
                color = SansColors.Gray400,
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 8.dp),
            )

            Spacer(Modifier.height(16.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(4.dp),
            ) {
                allGraphs.forEachIndexed { i, g ->
                    ThumbnailButton(
                        graph = g,
                        selected = i == currentIndex,
                        onClick = { onIndexChange(i) },
                        modifier = Modifier.weight(1f),
                    )
                }
            }

            Spacer(Modifier.height(16.dp))

            CategoryLegendCard()
        }
    }
}

@Composable
private fun GraphsImmersiveLayout(
    graph: GraphConfig,
    currentIndex: Int,
    onIndexChange: (Int) -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxSize()
            .background(SansColors.White)
            .padding(8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        LandscapeGraphSidebar(
            graph = graph,
            navContent = {
                LandscapeFigureNavRow(
                    currentIndex = currentIndex,
                    onIndexChange = onIndexChange,
                )
            },
        )

        AspectFitGraphFrame(
            modifier = Modifier
                .weight(1f)
                .fillMaxHeight()
                .padding(start = 8.dp),
        ) {
            ZoomableGraphBox(modifier = Modifier.fillMaxSize()) {
                Sans347Graph(
                    config = graph,
                    fitContainer = true,
                    modifier = Modifier.fillMaxSize(),
                )
            }
        }
    }
}

@Composable
private fun GraphHeaderCard(
    graph: GraphConfig,
    currentIndex: Int,
    onIndexChange: (Int) -> Unit,
    compact: Boolean = false,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(
                if (compact) SansColors.White.copy(alpha = 0.92f) else SansColors.Gray50,
                RoundedCornerShape(12.dp),
            )
            .border(1.dp, SansColors.Gray200, RoundedCornerShape(12.dp))
            .padding(if (compact) 8.dp else 16.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        if (!compact) {
            Text(
                "${currentIndex + 1} of 9",
                fontSize = 14.sp,
                color = SansColors.Gray400,
                modifier = Modifier
                    .align(Alignment.End)
                    .background(SansColors.Gray200, RoundedCornerShape(999.dp))
                    .padding(horizontal = 8.dp, vertical = 4.dp),
            )
        }
        Text(
            graph.title,
            fontWeight = FontWeight.Bold,
            fontSize = if (compact) 14.sp else 18.sp,
            color = Color.Black,
            textAlign = TextAlign.Center,
        )
        if (graph.subtitle.isNotEmpty()) {
            Text(
                graph.subtitle,
                color = SansColors.PrimaryCyan,
                fontWeight = FontWeight.Medium,
                fontSize = if (compact) 12.sp else 14.sp,
                textAlign = TextAlign.Center,
            )
        }
        Row(
            modifier = Modifier.fillMaxWidth().padding(top = if (compact) 4.dp else 8.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                "Figure ${graph.id}${if (compact) " · ${currentIndex + 1}/9" else ""}",
                color = SansColors.Gray500,
                fontSize = if (compact) 12.sp else 14.sp,
            )
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                val canPrev = currentIndex > 0
                val canNext = currentIndex < 8
                Button(
                    onClick = { onIndexChange(currentIndex - 1) },
                    enabled = canPrev,
                    modifier = Modifier.alpha(if (canPrev) 1f else 0.3f),
                    colors = ButtonDefaults.outlinedButtonColors(contentColor = SansColors.Gray700),
                    border = ButtonDefaults.outlinedButtonBorder(enabled = canPrev),
                    contentPadding = PaddingValues(horizontal = 8.dp, vertical = 4.dp),
                ) {
                    Icon(Icons.AutoMirrored.Filled.KeyboardArrowLeft, null)
                    if (!compact) {
                        Text("Prev", fontSize = 12.sp)
                    }
                }
                Button(
                    onClick = { onIndexChange(currentIndex + 1) },
                    enabled = canNext,
                    modifier = Modifier.alpha(if (canNext) 1f else 0.3f),
                    colors = ButtonDefaults.outlinedButtonColors(contentColor = SansColors.Gray700),
                    border = ButtonDefaults.outlinedButtonBorder(enabled = canNext),
                    contentPadding = PaddingValues(horizontal = 8.dp, vertical = 4.dp),
                ) {
                    if (!compact) {
                        Text("Next", fontSize = 12.sp)
                    }
                    Icon(Icons.AutoMirrored.Filled.KeyboardArrowRight, null)
                }
            }
        }
    }
}

@Composable
private fun ThumbnailButton(
    graph: GraphConfig,
    selected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val abbr = when (graph.equipmentType) {
        "Pressure Vessels" -> "Vessels"
        "Steam Generator" -> "Steam Gen"
        else -> "Piping"
    }
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = modifier
            .border(
                width = 2.dp,
                color = if (selected) SansColors.PrimaryCyan else SansColors.Gray200,
                shape = RoundedCornerShape(8.dp),
            )
            .background(if (selected) SansColors.ThumbSelectedBg else SansColors.White, RoundedCornerShape(8.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 4.dp, vertical = 6.dp),
    ) {
        Text(
            "Fig ${graph.id}",
            fontWeight = FontWeight.Bold,
            fontSize = 9.sp,
            color = if (selected) SansColors.PrimaryCyan else SansColors.Gray700,
            textAlign = TextAlign.Center,
        )
        Text(
            abbr,
            fontSize = 7.sp,
            color = SansColors.Gray400,
            maxLines = 1,
            textAlign = TextAlign.Center,
        )
    }
}

@Composable
private fun CategoryLegendCard() {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(SansColors.White, RoundedCornerShape(12.dp))
            .border(1.dp, SansColors.Gray200, RoundedCornerShape(12.dp))
            .padding(16.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Text("Category Legend", fontWeight = FontWeight.Bold, color = Color.Black, textAlign = TextAlign.Center, modifier = Modifier.padding(bottom = 12.dp))
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            listOf(
                Triple("SEP", SansColors.Success, "Sound Engineering Practice"),
                Triple("I", SansColors.CategoryI, "Category I"),
                Triple("II", SansColors.CategoryII, "Category II"),
                Triple("III", SansColors.CategoryIII, "Category III"),
                Triple("IV", SansColors.CategoryIV, "Category IV"),
            ).forEach { (label, color, desc) ->
                Column(
                    modifier = Modifier.weight(1f),
                    horizontalAlignment = Alignment.CenterHorizontally,
                ) {
                    Text(
                        label,
                        color = SansColors.White,
                        fontWeight = FontWeight.Bold,
                        fontSize = 14.sp,
                        textAlign = TextAlign.Center,
                        modifier = Modifier
                            .fillMaxWidth()
                            .background(color, RoundedCornerShape(8.dp))
                            .padding(vertical = 8.dp),
                    )
                    Text(desc, fontSize = 10.sp, color = SansColors.Gray500, textAlign = TextAlign.Center, modifier = Modifier.padding(top = 4.dp))
                }
            }
        }
    }
}

File: iOS/SANS347/Views/Screens/GraphsScreen.swift
import SwiftUI

/// Graphs page, mirroring `GraphsScreen.kt` — browse all 9 figures.
struct GraphsScreen: View {
    @EnvironmentObject private var state: AppState
    var immersiveGraph: Bool = false

    private var currentIndex: Int { min(max(state.currentGraphIndex, 0), 8) }
    private var graph: GraphConfig { allGraphs[currentIndex] }

    var body: some View {
        if immersiveGraph {
            immersiveLayout
        } else {
            portraitLayout
        }
    }

    private var portraitLayout: some View {
        ScrollView {
            VStack(spacing: 16) {
                headerCard
                SansCard(padding: 8) {
                    ZoomableGraphBox {
                        Sans347GraphView(config: graph)
                            .aspectRatio(GRAPH_ASPECT_RATIO, contentMode: .fit)
                    }
                }
                applicationCard
                Text(graph.footerText)
                    .font(.system(size: 14))
                    .foregroundColor(SansColors.gray400)
                    .multilineTextAlignment(.center)
                    .frame(maxWidth: .infinity)
                thumbnailRow
                legendCard
            }
            .maxContentWidth(896)
            .padding(.horizontal, 16)
            .padding(.bottom, 100)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(SansColors.white)
    }

    /// Landscape immersive layout (chrome hidden) — graph fills, controls in a
    /// compact sidebar. Mirrors `GraphsImmersiveLayout` in `GraphsScreen.kt`.
    private var immersiveLayout: some View {
        HStack(spacing: 8) {
            VStack(alignment: .leading, spacing: 8) {
                Text(graph.title).font(.system(size: 14, weight: .bold)).foregroundColor(.black)
                if !graph.subtitle.isEmpty {
                    Text(graph.subtitle).font(.system(size: 12, weight: .medium)).foregroundColor(SansColors.primaryCyan)
                }
                Text("Figure \(graph.id) · \(currentIndex + 1)/9").font(.system(size: 12)).foregroundColor(SansColors.gray500)
                HStack(spacing: 6) {
                    navButton(systemName: "chevron.left", label: "Prev", enabled: currentIndex > 0) {
                        state.setCurrentGraphIndex(currentIndex - 1)
                    }
                    navButton(label: "Next", systemName: "chevron.right", enabled: currentIndex < 8) {
                        state.setCurrentGraphIndex(currentIndex + 1)
                    }
                }
                ScrollView {
                    Text(graph.applicationText)
                        .font(.system(size: 11)).foregroundColor(SansColors.gray700)
                }
                Spacer(minLength: 0)
            }
            .frame(width: 200)

            ZoomableGraphBox {
                Sans347GraphView(config: graph)
                    .aspectRatio(GRAPH_ASPECT_RATIO, contentMode: .fit)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
        .padding(8)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(SansColors.white)
    }

    private var headerCard: some View {
        VStack(spacing: 0) {
            Text("\(currentIndex + 1) of 9")
                .font(.system(size: 14)).foregroundColor(SansColors.gray400)
                .padding(.horizontal, 8).padding(.vertical, 4)
                .background(SansColors.gray200)
                .clipShape(Capsule())
                .frame(maxWidth: .infinity, alignment: .trailing)
            Text(graph.title).font(.system(size: 18, weight: .bold)).foregroundColor(.black)
                .multilineTextAlignment(.center)
            if !graph.subtitle.isEmpty {
                Text(graph.subtitle).font(.system(size: 14, weight: .medium)).foregroundColor(SansColors.primaryCyan)
            }
            HStack {
                Text("Figure \(graph.id)").font(.system(size: 14)).foregroundColor(SansColors.gray500)
                Spacer()
                HStack(spacing: 6) {
                    navButton(systemName: "chevron.left", label: "Prev", enabled: currentIndex > 0) {
                        state.setCurrentGraphIndex(currentIndex - 1)
                    }
                    navButton(label: "Next", systemName: "chevron.right", enabled: currentIndex < 8) {
                        state.setCurrentGraphIndex(currentIndex + 1)
                    }
                }
            }
            .padding(.top, 8)
        }
        .padding(16)
        .frame(maxWidth: .infinity)
        .background(SansColors.gray50)
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(SansColors.gray200, lineWidth: 1))
    }

    private func navButton(systemName: String? = nil, label: String, enabled: Bool, action: @escaping () -> Void) -> some View {
        navButton(label: label, leadingSystemName: systemName, trailingSystemName: nil, enabled: enabled, action: action)
    }

    private func navButton(label: String, systemName: String, enabled: Bool, action: @escaping () -> Void) -> some View {
        navButton(label: label, leadingSystemName: nil, trailingSystemName: systemName, enabled: enabled, action: action)
    }

    private func navButton(label: String, leadingSystemName: String?, trailingSystemName: String?, enabled: Bool, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            HStack(spacing: 4) {
                if let leadingSystemName { Image(systemName: leadingSystemName) }
                Text(label).font(.system(size: 12))
                if let trailingSystemName { Image(systemName: trailingSystemName) }
            }
            .foregroundColor(SansColors.gray700)
            .padding(.horizontal, 8).padding(.vertical, 4)
            .overlay(RoundedRectangle(cornerRadius: 8).stroke(SansColors.gray300, lineWidth: 1))
        }
        .buttonStyle(.plain)
        .opacity(enabled ? 1 : 0.3)
        .disabled(!enabled)
    }

    private var applicationCard: some View {
        Text(graph.applicationText)
            .font(.system(size: 14)).foregroundColor(SansColors.gray700)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
            .padding(16)
            .background(SansColors.white)
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(SansColors.gray200, lineWidth: 1))
    }

    private var thumbnailRow: some View {
        HStack(spacing: 4) {
            ForEach(Array(allGraphs.enumerated()), id: \.element.id) { index, g in
                thumbnail(g, selected: index == currentIndex) {
                    state.setCurrentGraphIndex(index)
                }
            }
        }
    }

    private func thumbnail(_ g: GraphConfig, selected: Bool, action: @escaping () -> Void) -> some View {
        let abbr: String
        switch g.equipmentType {
        case "Pressure Vessels": abbr = "Vessels"
        case "Steam Generator": abbr = "Steam Gen"
        default: abbr = "Piping"
        }
        return Button(action: action) {
            VStack(spacing: 2) {
                Text("Fig \(g.id)")
                    .font(.system(size: 9, weight: .bold))
                    .foregroundColor(selected ? SansColors.primaryCyan : SansColors.gray700)
                Text(abbr).font(.system(size: 7)).foregroundColor(SansColors.gray400).lineLimit(1)
            }
            .frame(maxWidth: .infinity)
            .padding(.horizontal, 4).padding(.vertical, 6)
            .background(selected ? SansColors.thumbSelectedBg : SansColors.white)
            .clipShape(RoundedRectangle(cornerRadius: 8))
            .overlay(RoundedRectangle(cornerRadius: 8).stroke(selected ? SansColors.primaryCyan : SansColors.gray200, lineWidth: 2))
        }
        .buttonStyle(.plain)
    }

    private var legendCard: some View {
        SansCard {
            VStack(spacing: 12) {
                Text("Category Legend").font(.system(size: 16, weight: .bold)).foregroundColor(.black)
                HStack(spacing: 8) {
                    legendItem("SEP", SansColors.success, "Sound Engineering Practice")
                    legendItem("I", SansColors.categoryI, "Category I")
                    legendItem("II", SansColors.categoryII, "Category II")
                    legendItem("III", SansColors.categoryIII, "Category III")
                    legendItem("IV", SansColors.categoryIV, "Category IV")
                }
            }
            .frame(maxWidth: .infinity)
        }
    }

    private func legendItem(_ label: String, _ color: Color, _ desc: String) -> some View {
        VStack(spacing: 4) {
            Text(label)
                .font(.system(size: 14, weight: .bold)).foregroundColor(.white)
                .frame(maxWidth: .infinity).padding(.vertical, 8)
                .background(color).clipShape(RoundedRectangle(cornerRadius: 8))
            Text(desc).font(.system(size: 10)).foregroundColor(SansColors.gray500).multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity)
    }
}
